src/twisted/web/server.py

```diff
@@ -507,7 +507,7 @@ def _prePathURL(self, prepath):
             hostport = ':%d' % port
         prefix = networkString('http%s://%s%s/' % (
             self.isSecure() and 's' or '',
-            nativeString(self.getRequestHostname()),
+            nativeString(self.getRequestHost()),
```

Comment:
The immediate conversion of the network string to a native string here is hinting that the method should just return a native string. ;-)